lib/index.js

```diff
@@ -1430,6 +1433,65 @@ Auth0.prototype.getConnections = function (callback) {
 };
 
 /**
+ * Send SMS to do passwordless authentication
+ *
+ * Examples:
+ *
+ *     auth0.requestSMSCode(apiToken, phoneNumber, function (err, result) {
+ *       if (err) return console.log(err.message);
+ *       console.log(result);
+ *     });
+ *
+ * @method requestSMSCode
+ * @param {Object} options
+ * @param {Function} callback
+ */
+
+Auth0.prototype.requestSMSCode = function (options, callback) {
+  if ('object' !== typeof options) {
+    throw new Error('An options object is required');
+  }
+  if ('function' !== typeof callback) {
+    throw new Error('A callback function is required');
+  }
+
+  assert_required(options, 'apiToken');
+  assert_required(options, 'phoneNumber');
+
+  var apiToken = options.apiToken;
+  var phoneNumber = options.phoneNumber;
+
+  return reqwest({
+    url:          'https://' + this._domain + '/api/v2/users',
+    method:       'post',
+    type:         'json',
+    crossOrigin:  true,
+    headers:      {
+      Authorization: 'Bearer ' + apiToken
+    },
+    data:         {
+      phone_number:   phoneNumber,
+      connection:     'sms',
+      email_verified: false
+    }
+  })
+  .fail(function (err) {
+    try {
```

Comment:
There are a couple of failure corner cases handled by this function: https://github.com/auth0/auth0.js/blob/master/lib/index.js#L84-L120 . Do you think we can re use that function here?